File: tests/machine/x/clj/match_expr.clj
(ns main)

(declare x label)

(defn -main []
  (def x 2) ;; int
  (def label (let [t x]
  (cond
    (= t 1) "one"
    (= t 2) "two"
    (= t 3) "three"
    :else "unknown"
  ))) ;; string
  (println label)
)

(-main)

File: tests/transpiler/x/pl/match_expr.pl
:- initialization(main).
:- style_check(-singleton).

main :-
    X is 2,
    (2 =:= 1 -> Label = "one" ; (2 =:= 2 -> Label = "two" ; (2 =:= 3 -> Label = "three" ; Label = "unknown"))),
    writeln(Label).
